app/models/machine.rb

```diff
@@ -1,5 +1,20 @@
 class Machine < ApplicationRecord
   validates_presence_of :location
-  
+  has_many :machine_snacks
+  has_many :snacks, through: :machine_snacks
   belongs_to :owner
+
+  def average_snack_price
+    snacks.average(:price)
+  end
+
+  def display_average_snack_price
+    if average_snack_price
+      "$#{sprintf('%.2f', average_snack_price)}"
+    end
```

Comment:
You can look into an actionview helper called `number_to_currency` that will take a number, and turn it into a dolalr amount.